src/infrasys/system.py

```diff
@@ -1094,6 +1095,19 @@ def merge_system(self, other: "System") -> None:
 
     # TODO: add delete methods that (1) don't raise if not found and (2) don't return anything?
 
+    def convert_storage(self, replace=True, **kwargs) -> TimeSeriesStorageBase | None:
+        """ 
+        Converts the time series storage medium.
+
+        Parameters
+        ----------
+        replace: bool
+            if True, the underlying storage will be replaced with new storage
+            otherwise, the new storage object is returned
+
+        """
+        return self._time_series_mgr.convert_storage(**kwargs, replace=replace)
```

Comment:
would it makes sense to return a copy of time_series_mgr with new storage? or the storage itself? or only implement this method for System with replace=True?

Reply:
The more I think about this the more I think that there is no use case for replace = False. Current thoughts:
1. Replace the storage object. Return nothing. Leave the data in the tmp dir or maybe delete it.
2. Leverage this feature in the deserialize workflow. User starts with in-memory, serializes to disk (gets Arrow storage), then deserializes and wants it converted back to memory. The user never sees or has access to the intermediate Arrow storage.

What do you think?

Reply:
I agree. The replace=False option seems like it's trying to be future-proof, but it also seems like we really don't, and probably won't ever need it.